test/integration/pack/test_pack.py

```diff
@@ -990,3 +1002,266 @@ def test_verbosity(cartridge_cmd, project_without_dependencies, pack_format):
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    if pack_format == 'deb':
+        deps = (
+            "cool_dependency_01 (>= 1.2)",
+            "cool_dependency_01 (<< 3)",
+            "cool_dependency_02 (= 2.5)",
+            "cool_dependency_03"
+        )
+
+        if not tarantool_enterprise_is_used():
+            deps += (
+                "tarantool (>= {})".format(tarantool_versions["min"]["deb"]),
+                "tarantool (<< {})".format(tarantool_versions["max"]["deb"]),
+            )
+
+        assert_dependencies_deb(find_archive(tmpdir, project.name, 'deb'), deps, tmpdir)
+    else:
+        deps = (
+            ("cool_dependency_01", 0x08 | 0x04, "1.2"),  # >=
+            ("cool_dependency_01", 0x02, "3"),  # <
+            ("cool_dependency_02", 0x08, "2.5"),  # =
+            ("cool_dependency_03", 0, "")
+        )
+
+        if not tarantool_enterprise_is_used():
+            deps += (
+                ("tarantool", 0x08 | 0x04, tarantool_versions["min"]["rpm"]),  # >=
+                ("tarantool", 0x02, tarantool_versions["max"]["rpm"]),
+            )
+
+        assert_dependencies_rpm(find_archive(tmpdir, project.name, 'rpm'), deps)
+
+
+@pytest.mark.parametrize('pack_format', ['deb', 'rpm'])
+def test_custom_dependencies_file(cartridge_cmd, project_without_dependencies, pack_format, tarantool_versions, tmpdir):
+    project = project_without_dependencies
+
+    deps_filepath = os.path.join(tmpdir, "deps.txt")
+    with open(deps_filepath, "w") as f:
+        f.write("dependency01 >= 1.2, < 3 \n" +
+                "  dependency02 == 2.5\n" +
+                "\tdependency03  ")
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--deps-file", deps_filepath,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    if pack_format == 'deb':
+        deps = (
+            "dependency01 (>= 1.2)",
+            "dependency01 (<< 3)",
+            "dependency02 (= 2.5)",
+            "dependency03"
+        )
+
+        if not tarantool_enterprise_is_used():
+            deps += (
+                "tarantool (>= {})".format(tarantool_versions["min"]["deb"]),
+                "tarantool (<< {})".format(tarantool_versions["max"]["deb"]),
+            )
+
+        assert_dependencies_deb(find_archive(tmpdir, project.name, 'deb'), deps, tmpdir)
+    else:
+        deps = (
+            ("dependency01", 0x08 | 0x04, "1.2"),  # >=
+            ("dependency01", 0x02, "3"),  # <
+            ("dependency02", 0x08, "2.5"),  # =
+            ("dependency03", 0, "")
+        )
+
+        if not tarantool_enterprise_is_used():
```

Comment:
Seems like a function (not copy-paste). In fact, it can be moved to `assert_dependencies_*` functions.